src/Expression.php

```diff
@@ -77,6 +77,13 @@ class Expression implements \ArrayAccess, \IteratorAggregate
      */
     public $connection;
 
+    /**
+     * Wrap the expression in parenthesis when consumed by another expression or not.
+     *
+     * @var bool
+     * */
+    public $consumeWrappedInParenthesis = false;
```

Comment:
why past tense?

Reply:
Dis is how you say it: something needs to be wrapped or not. "Wrap" is the act of doing so

Reply:
yes, what grades did you have?

somethings needs to be wrapped, yes

consumeWraps - when expr is beiing consumed, consume wraps

PS: not a native speaker, but always A+ in Czech :D

Reply:
No, I wrote it in English, not in Czech :)

It is written correctly in English.

It is meant not what is done to it (wrap as verb) but how it is consumed (wrapped as adjective).
The verb is "consume".
"Consume the expression wrapped in paranthesis."

Reply:
You mean "it is wrapped", I meant "consume wraps" - when it arrives to consume, is it already wrapped?

If yes, I think you are right. If no, and concume does it, then I think, wrapped is not correct.

Reply:
I understand you mean "consume method wraps it" but this is not an indicator of what the method has done to it but how it should be consumed.
"Consumption" is performed by the method.
So the method checks if expression should be consumed wrapped or not and performs accordingly.

Reply:
I got it - consume once wrapped :))

but I find it confusing, as we asks consume to do it so

that is why named `allow...` previously... 3rd person feedback would be welcomed, @DarkSide666, would it be you?

Reply:
I think "allow" is not intuitive as the wrapping is strictly determined only by this setting. Allow implies it is optional - it can be allowed but not done, which is not the case. 
Let's see what @DarkSide666 will say.

Reply:
There was something about UNION when it was not allowed to wrap if I remember...

Reply:
Yes, queries are wrapped by default when consumed. But when union is used this can be switched off as some platforms don't support wrapped union queries.
Now I added that where clauses from an expression are also wrapped in paranthesis.

Let's merge this and move on to next issue...

Reply:
In case of union it has to be `SELECT * FROM t1 UNION SELECT * FROM t2` without wrapping select queries in parenthesis. I don't remember exactly now which engine didn't support `(select ...) union (select ...)`, but it was Oracle or PostGRE i guess.

Reply:
It is Sqlite as per code comments
Now we use this same feature to wrap expressions used as where conditions in ()